Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoASA.yaml

```diff
@@ -0,0 +1,576 @@
+          , "112","VRRP"
+          , "113","PGM"
+          , "114",""
+          , "115","L2TP"
+          , "116","DDX"
+          , "117","IATP"
+          , "118","STP"
+          , "119","SRP"
+          , "120","UTI"
+          , "121","SMP"
+          , "122","SM (deprecated)"
+          , "123","PTP"
+          , "124","ISIS over IPv4"
+          , "125","FIRE"
+          , "126","CRTP"
+          , "127","CRUDP"
+          , "128","SSCOPMCE"
+          , "129","IPLT"
+          , "130","SPS"
+          , "131","PIPE"
+          , "132","SCTP"
+          , "133","FC"
+          , "134","RSVP-E2E-IGNORE"
+          , "135","Mobility Header"
+          , "136","UDPLite"
+          , "137","MPLS-in-IP"
+          , "138","manet"
+          , "139","HIP"
+          , "140","Shim6"
+          , "141","WESP"
+          , "142","ROHC"
+          , "143","Ethernet"
+          , "253",""
+          , "254",""
+          , "255","Reserved"
+      ];
+      let NWParser = (
+            starttime:datetime=datetime(null), 
+            endtime:datetime=datetime(null),
+            srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+            dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+            ipaddr_has_any_prefix:dynamic=dynamic([]),
+            dstportnumber:int=int(null), 
+            hostname_has_any:dynamic=dynamic([]), 
+            dvcaction:dynamic=dynamic([]), 
+            eventresult:string='*', 
+            disabled:bool=false)
+        { 
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let allLogs = CommonSecurityLog
+                | where not(disabled)
+                | where (isnull(starttime) or TimeGenerated >= starttime) and (isnull(endtime) or TimeGenerated <= endtime)
+                | where DeviceVendor == "Cisco" and DeviceProduct == "ASA"
+                | where (isnull(dstportnumber) or (DestinationPort == dstportnumber)) or Message has tostring(dstportnumber)
+                | extend temp_isSrcMatch=has_any_ipv4_prefix(SourceIP,src_or_any), 
+                                temp_isDstMatch=has_any_ipv4_prefix(DestinationIP,dst_or_any)
+                | extend ASimMatchingIpAddr = case(
+                                                array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-", // match not requested
+                                                (temp_isSrcMatch and temp_isDstMatch), "Both", // has to be checked before the individual 
+                                                temp_isSrcMatch, "SrcIpAddr",
+                                                temp_isDstMatch, "DstIpAddr",
+                                                "No match"
+                                            )
+                | where ASimMatchingIpAddr != "No match"
+                | where ((array_length(dvcaction) == 0) or SimplifiedDeviceAction has_any (dvcaction))
```

Comment:
At least for 302014, this condition is incorrect. Is it correct for all others? We need to find a way to do it only when relevant. It is not an easy one and requires some thinking about overall structure.

Reply:
Updated